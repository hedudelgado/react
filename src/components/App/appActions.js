import { HeaderInterface, InvalidSessionErrorCodeMap, InvalidSessionHelper } from 'mega-code';
import {
  getCountries,
  getNewsletterRegions,
  getNewsletterPreferences,
  getProfile,
  updateProfile,
  deleteCustomer,
} from '../../utils/api';

// actions
export const USER_LOGGED_IN_SUCCESSFULLY = 'USER_LOGGED_IN_SUCCESSFULLY';
export const COUNTRIES_RECEIVED = 'COUNTRIES_RECEIVED';
export const PROFILE_RECEIVED = 'PROFILE_RECEIVED';
export const REGIONS_RECEIVED = 'REGIONS_RECEIVED';
export const NEWSLETTER_RECEIVED = 'NEWSLETTER_RECEIVED';
export const SET_EDIT_VIEW = 'SET_EDIT_VIEW';
export const SET_NOTIFICATION_MESSAGE = 'SET_NOTIFICATION_MESSAGE';
export const PROFILE_UPDATED = 'PROFILE_UPDATED';
export const NEWSLETTER_UPDATED = 'NEWSLETTER_UPDATED';
export const RESET_NEWSLETTER_INFO = 'RESET_NEWSLETTER_INFO';
export const RESET_USER_INFO = 'RESET_USER_INFO';
// views
export const DEFAULT_VIEW = undefined;
export const USER_DETAILS_VIEW = 'USER_DETAILS_VIEW';
export const PASSWORD_VIEW = 'PASSWORD_VIEW';
export const PAYMENT_CARD_VIEW = 'PAYMENT_CARD_VIEW';
export const ADD_GUEST_VIEW = 'ADD_GUEST_VIEW';
export const REGULAR_GUESTS_VIEW = 'REGULAR_GUESTS_VIEW';
export const NEWSLETTER_VIEW = 'NEWSLETTER_VIEW';
export const ROOM_REQUIREMENTS_VIEW = 'ROOM_REQUIREMENTS_VIEW';
export const EXTRAS_PREFERENCES_VIEW = 'EXTRAS_PREFERENCES_VIEW';
export const MEMORABLE_WORD_VIEW = 'MEMORABLE_WORD_VIEW';
// errors
export const PROFILE_ERROR = 'PROFILE_ERROR';
export const PASSWORD_ERROR = 'PASSWORD_ERROR';
export const NEWSLETTER_ERROR = 'NEWSLETTER_ERROR';
export const PAYMENT_CARD_ERROR = 'PAYMENT_CARD_ERROR';
export const USER_DETAILS_ERROR = 'USER_DETAILS_ERROR';
export const REGULAR_GUESTS_ERROR = 'REGULAR_GUESTS_ERROR';
export const MEMORABLE_WORD_ERROR = 'MEMORABLE_WORD_ERROR';
export const ROOM_REQUIREMENTS_ERROR = 'ROOM_REQUIREMENTS_ERROR';
export const EXTRAS_PREFERENCES_ERROR = 'EXTRAS_PREFERENCES_ERROR';

const checkForInvalidSession = (error, config, dispatch) => {
  if (InvalidSessionHelper
    .isSessionExpired(error, config)) {
    dispatch({
      type: RESET_USER_INFO,
    });
    dispatch({
      type: RESET_NEWSLETTER_INFO,
    });
    HeaderInterface.userLogOut(HeaderInterface.LOG_OUT_REASONS.SESSION_EXPIRED);
  }
};

export const getSettings = (email, sessionId, business) => (dispatch, getState) => {
  getProfile(email, sessionId, business).then((profile) => {
    dispatch({
      type: PROFILE_RECEIVED,
      profile,
    });
  }).catch((error) => {
    checkForInvalidSession(error, InvalidSessionErrorCodeMap.ACCOUNT.invalidSession, dispatch);
    dispatch({
      type: PROFILE_ERROR,
      error,
    });
  });
  getCountries()
    .then((countries) => {
      dispatch({
        type: COUNTRIES_RECEIVED,
        countries,
      });
    });
  if (getState().application.showEmailPreferences) {
    getNewsletterRegions()
      .then((regions) => {
        dispatch({
          type: REGIONS_RECEIVED,
          regions,
        });
      });

    getNewsletterPreferences(email)
      .then((newsletter) => {
        dispatch({
          type: NEWSLETTER_RECEIVED,
          newsletter,
        });
      })
      .catch(e => dispatch({
        type: NEWSLETTER_ERROR,
        error: e,
      }));
  }
};

export const updateUserProfile = payload => dispatch => updateProfile(payload).catch((error) => {
  checkForInvalidSession(error, InvalidSessionErrorCodeMap.ACCOUNT.invalidSession, dispatch);
  return Promise.reject(error);
});

export const deleteUserProfile = payload => dispatch => deleteCustomer(payload).catch((error) => {
  checkForInvalidSession(error, InvalidSessionErrorCodeMap.DELETE_PROFILE.invalidSession, dispatch);
  return Promise.reject(error);
});

export default {
  getSettings,
};
