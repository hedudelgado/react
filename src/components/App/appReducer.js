import {
  SET_NOTIFICATION_MESSAGE, SET_EDIT_VIEW, DEFAULT_VIEW,
} from './appActions';

const defaultState = {
  editViewName: DEFAULT_VIEW,
};

export default (state = defaultState, action) => {
  switch (action.type) {
    case SET_EDIT_VIEW:
      return {
        ...state,
        editViewName: action.editViewName,
      };
    case SET_NOTIFICATION_MESSAGE:
      return {
        ...state,
        notificationMessage: action.notificationMessage,
      };
    default:
      return state;
  }
};
