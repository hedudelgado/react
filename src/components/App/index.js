export { default } from './App';

export { default as AppActions } from './appActions';

export { default as AppContainer } from './AppContainer';

export { default as modalReducer } from './modalReducer';
