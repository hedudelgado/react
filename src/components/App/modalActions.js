export const MODAL_OPEN = 'MODAL_OPEN';
export const MODAL_CLOSE = 'MODAL_CLOSE';

export const openModal = content => ({
  type: MODAL_OPEN,
  content,
});

export const closeModal = () => ({
  type: MODAL_CLOSE,
});
