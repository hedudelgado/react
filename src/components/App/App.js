import React, { Component } from 'react';
import {
  array,
  object,
  func,
  bool,
  node,
} from 'prop-types';
import {
  PASSWORD_VIEW,
  PAYMENT_CARD_VIEW,
  USER_DETAILS_VIEW,
  ROOM_REQUIREMENTS_VIEW,
  EXTRAS_PREFERENCES_VIEW,
} from './appActions';
import { ComponentTogglerContainer } from '../ComponentToggler';
import UserProfile from '../UserProfile';
import Password from '../Password';
import UserProfileFormContainer from '../UserProfileForm';
import PasswordFormContainer from '../PasswordForm';
import { PaymentCardContainer } from '../PaymentCard';
import { RegularGuestsContainer } from '../RegularGuests';
import { NewsletterContainer } from '../Newsletter';
import Modal from '../../layout/Modal';
import PaymentCardFormContainer from '../PaymentCardForm';
import { DeleteCustomerContainer } from '../DeleteCustomer';
import RoomRequirements from '../RoomRequirements';
import RoomRequirementsFormContainer from '../RoomRequirementsForm';
import ExtrasPreferences from '../ExtrasPreferences';
import ExtrasPreferencesFormContainer from '../ExtrasPreferencesForm';
import BaseStyle from '../../layout/BaseStyle';
import MemorableWord from '../MemorableWord';
import MoveAccount from '../MoveAccount';
import {
  STAYER, SELF, managePermissionFlag, USER_LOGGED_IN_EVENT,
} from '../../config.json';
import ManagePermission from '../ManagePermission';

class App extends Component {
  constructor(props) {
    super(props);
    this.handleLoggedIn = this.handleLoggedIn.bind(this);
  }

  componentDidMount() {
    window.addEventListener('message', this.handleLoggedIn);
  }

  componentWillUnmount() {
    window.removeEventListener('message', this.handleLoggedIn);
  }

  async handleLoggedIn(event) {
    if (event.data.action === USER_LOGGED_IN_EVENT) {
      const { getSettings } = this.props;
      const { contactDetail: { email }, sessionId, business } = JSON.parse(event.data.profile);
      getSettings(email, sessionId, !!business);
    }
  }

  render() {
    const {
      countries,
      dictionary,
      profile,
      regions,
      newsletter,
      displayModal,
      handleModalClose,
      modalContent,
      userDataReceived,
      application: {
        showEmailPreferences,
      },
    } = this.props;

    const {
      business,
      business: {
        accessLevel = '',
        tetheredGuid = '',
      } = {},
    } = profile;

    const appReady = userDataReceived
      && (!showEmailPreferences || !!regions.length)
      && !!countries.length
      && (!showEmailPreferences || !!Object.keys(newsletter).length)
      && !!Object.keys(dictionary).length
      && !!Object.keys(profile).length;

    const showExtrasPreferences = !(accessLevel === SELF || accessLevel === STAYER);

    return (
      <div data-test="AccountSettings">
        {appReady
        && (
          <BaseStyle className="wb-typography app-container">
            <ComponentTogglerContainer
              viewName={USER_DETAILS_VIEW}
              view={UserProfile}
              edit={UserProfileFormContainer}
            />
            <ComponentTogglerContainer
              viewName={PASSWORD_VIEW}
              view={Password}
              edit={PasswordFormContainer}
            />
            <ComponentTogglerContainer
              viewName={PAYMENT_CARD_VIEW}
              view={PaymentCardContainer}
              edit={PaymentCardFormContainer}
            />
            {managePermissionFlag && (
              <ManagePermission dictionary={dictionary} />
            )}
            <RegularGuestsContainer />
            {showEmailPreferences && (
              <NewsletterContainer />
            )}
            <ComponentTogglerContainer
              viewName={ROOM_REQUIREMENTS_VIEW}
              view={RoomRequirements}
              edit={RoomRequirementsFormContainer}
            />
            {showExtrasPreferences && (
              <ComponentTogglerContainer
                viewName={EXTRAS_PREFERENCES_VIEW}
                view={ExtrasPreferences}
                edit={ExtrasPreferencesFormContainer}
              />
            )}
            {!!tetheredGuid && (
              <MemorableWord />
            )}
            {business && !tetheredGuid && (
              <MoveAccount />
            )}
            {!business && (
            <DeleteCustomerContainer />
            )}
            {displayModal && <Modal data-test="Modal" display content={modalContent} handleClose={handleModalClose} />}
          </BaseStyle>
        )
      }
      </div>
    );
  }
}

App.propTypes = {
  countries: array.isRequired,
  dictionary: object.isRequired,
  profile: object.isRequired,
  getSettings: func.isRequired,
  displayModal: bool.isRequired,
  handleModalClose: func.isRequired,
  modalContent: node,
  newsletter: object.isRequired,
  regions: array.isRequired,
  userDataReceived: bool.isRequired,
  application: object.isRequired,
};

export default App;
