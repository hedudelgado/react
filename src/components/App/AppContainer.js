import { connect } from 'react-redux';
import { getSettings as getSettingsAction } from './appActions';
import { closeModal } from './modalActions';

import App from './App';

const mapStateToProps = ({
  modal: {
    display: displayModal,
    content: modalContent,
  },
  dictionary,
  config,
  countries,
  regions,
  newsletter,
  application,
  userProfile: {
    userDataReceived,
    profile,
  },
}) => ({
  profile,
  displayModal,
  modalContent,
  dictionary,
  config,
  countries,
  regions,
  newsletter,
  application,
  userDataReceived,
});

const mapDispatchToProps = dispatch => ({
  getSettings:
    (email, sessionId, business) => dispatch(getSettingsAction(email, sessionId, business)),
  handleModalClose: () => dispatch(closeModal()),
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
