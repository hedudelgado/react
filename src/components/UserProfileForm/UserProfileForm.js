import React from 'react';
import {
  func,
  object,
  array,
} from 'prop-types';
import Select from '../../layout/Select';
import Button from '../../layout/Button';
import Input from '../../layout/Input';
import { DEFAULT_VIEW } from '../App/appActions';
import { selectedFlagImage } from '../../utils/selectedFlagImage';
import AddressForm from '../AddressForm';

const UserProfileForm = ({
  profile,
  countries,
  dictionary,
  setEditView,
  handleBlur,
  handleChange,
  handleSubmit,
  touched,
  values,
  errors,
  setFieldValue,
}) => {
  const {
    title,
    firstName,
    lastName,
    email,
    dialingCountryCode,
    mobile,
    telephone,
  } = values;

  const guestTitleOptions = dictionary['profile.form.guesttitle.list']
    .split(',')
    .map(guestTitle => (
      <option key={guestTitle} value={guestTitle}>{guestTitle}</option>
    ));

  const countryOptions = code => countries.map((country) => {
    const label = code === country.countryCode ? country.dialingCode : `${country.countryLegend} (${country.dialingCode})`;
    return (
      <option
        label={label}
        value={country.countryCode}
        key={country.countryCode}
      >
        {label}
      </option>
    );
  });

  return (
    <form
      data-test="user-profile-form"
      name="user-profile-form"
      onSubmit={handleSubmit}
    >
      <Select
        onBlur={handleBlur}
        onChange={handleChange}
        value={title}
        name="title"
        id="title"
        placeholder={dictionary['profile.form.guesttitle']}
        touched={touched.title}
      >
        {guestTitleOptions}
      </Select>
      <Input
        onBlur={handleBlur}
        onChange={handleChange}
        value={firstName}
        type="text"
        name="firstName"
        id="firstName"
        placeholder={dictionary['profile.form.firstname']}
        error={dictionary[errors.firstName]}
        touched={touched.firstName}
      />
      <Input
        onBlur={handleBlur}
        onChange={handleChange}
        value={lastName}
        type="text"
        name="lastName"
        id="lastName"
        placeholder={dictionary['profile.form.lastname']}
        error={dictionary[errors.lastName]}
        touched={touched.lastName}
      />
      <Input
        value={email}
        type="text"
        name="email"
        id="email"
        placeholder={dictionary['profile.form.email']}
        disabled="disabled"
      />
      <Select
        onBlur={handleBlur}
        onChange={handleChange}
        name="dialingCountryCode"
        id="dialingCountryCode"
        value={dialingCountryCode}
        placeholder={dictionary['profile.form.dialingcode']}
        flagImg={selectedFlagImage(dialingCountryCode, countries)}
        touched={touched.dialingCountryCode}
      >
        {countryOptions(dialingCountryCode)}
      </Select>
      <Input
        onBlur={handleBlur}
        onChange={handleChange}
        value={mobile}
        type="text"
        name="mobile"
        id="mobile"
        placeholder={dictionary['profile.form.mobile']}
        error={dictionary[errors.mobile]}
        touched={touched.mobile}
      />
      <Input
        onBlur={handleBlur}
        onChange={handleChange}
        value={telephone}
        type="text"
        name="telephone"
        id="telephone"
        placeholder={dictionary['profile.form.telephone']}
        error={dictionary[errors.telephone]}
        touched={touched.telephone}
      />
      <AddressForm
        profile={profile}
        dictionary={dictionary}
        countries={countries}
        values={values}
        setFieldValue={setFieldValue}
        errors={errors}
        touched={touched}
        handleBlur={handleBlur}
        handleChange={handleChange}
      />
      <Button
        data-test="save-changes"
        className="wb-push--bottom"
        color="primary"
        type="submit"
      >
        {dictionary['profile.button.save']}
      </Button>
      <Button
        data-test="closeForm"
        color="default"
        variant="text"
        className="underline block"
        fullWidth
        onClick={() => setEditView(DEFAULT_VIEW)}
      >
        {dictionary['profile.button.cancel']}
      </Button>
    </form>
  );
};

UserProfileForm.propTypes = {
  profile: object.isRequired,
  countries: array.isRequired,
  dictionary: object.isRequired,
  setEditView: func.isRequired,
  handleBlur: func.isRequired,
  handleChange: func.isRequired,
  handleSubmit: func.isRequired,
  touched: object.isRequired,
  values: object.isRequired,
  errors: object.isRequired,
  setFieldValue: func.isRequired,
};

export default UserProfileForm;
