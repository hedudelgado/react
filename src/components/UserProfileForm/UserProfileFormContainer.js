import React, { Component } from 'react';
import {
  object,
  func,
  array,
} from 'prop-types';
import { Formik } from 'formik';
import BaseStyle from '../../layout/BaseStyle';
import Typography from '../../layout/Typography';
import Notification from '../../layout/Notification';
import UserProfileForm from './UserProfileForm';
import {
  validateFirstName,
  validateLastName,
  validatePhoneNumber,
  validatePostCode,
  validateCompanyName,
  validateAddressLine,
} from '../../utils/validations';
import { matchDialingCode } from '../../utils/matchDialingCode';
import { DEFAULT_VIEW, USER_DETAILS_VIEW, USER_DETAILS_ERROR } from '../App/appActions';
import { BUSINESS } from '../../config.json';

const validateUserProfileForm = ({
  firstName,
  lastName,
  mobile,
  telephone,
  line1,
  line2,
  line3,
  line4,
  postCode,
  countryCode,
  companyName,
  type,
}) => {
  const errors = {};

  errors.firstName = validateFirstName(firstName);
  errors.lastName = validateLastName(lastName);
  errors.postCode = validatePostCode(postCode, countryCode);
  errors.line1 = validateAddressLine(line1, true);
  errors.line2 = validateAddressLine(line2, false);
  errors.line3 = validateAddressLine(line3, false);
  errors.line4 = validateAddressLine(line4, false);

  if (!mobile && !telephone) {
    errors.mobile = 'profile.error.missingPhoneNumber';
    errors.telephone = 'profile.error.missingPhoneNumber';
  } else if (mobile && !telephone) {
    errors.mobile = validatePhoneNumber(mobile, 'profile.error.mobile');
  } else if (!mobile && telephone) {
    errors.telephone = validatePhoneNumber(telephone, 'profile.error.telephone');
  } else {
    errors.telephone = validatePhoneNumber(telephone, 'profile.error.telephone');
    errors.mobile = validatePhoneNumber(mobile, 'profile.error.mobile');
  }

  if (type === 'BUSINESS') {
    errors.companyName = validateCompanyName(companyName);
  }

  Object.keys(errors).forEach((error) => {
    if (errors[error] === undefined) delete errors[error];
  });

  return errors;
};

const processPhoneNumbers = (countries, profile) => {
  const {
    contactDetail: {
      telephone,
      mobile,
      address: { countryCode },
    },
  } = profile;

  let dialingCountryCode = '';

  let mobileProcessed = '';
  if (mobile) {
    const matchedDialingCountry = matchDialingCode(mobile, countries, countryCode);

    mobileProcessed = mobile;
    dialingCountryCode = matchedDialingCountry.countryCode;

    if (mobile.startsWith((matchedDialingCountry.dialingCode))) {
      mobileProcessed = mobile.replace(matchedDialingCountry.dialingCode, '');
    }
  }

  let telephoneProcessed = '';
  if (telephone) {
    const matchedDialingCountry = matchDialingCode(telephone, countries, countryCode);

    telephoneProcessed = telephone;
    dialingCountryCode = matchedDialingCountry.countryCode;

    if (telephone.startsWith((matchedDialingCountry.dialingCode))) {
      telephoneProcessed = telephone.replace(matchedDialingCountry.dialingCode, '');
    }
  }

  return {
    dialingCountryCode,
    mobile: mobileProcessed,
    telephone: telephoneProcessed,
  };
};

export default class UserProfileFormContainer extends Component {
  constructor(props) {
    super(props);
    this.handleSubmit = this.handleSubmit.bind(this);

    const { profile, countries } = props;

    const { mobile, telephone, dialingCountryCode } = processPhoneNumbers(countries, profile);

    this.mobile = mobile;
    this.telephone = telephone;
    this.dialingCountryCode = dialingCountryCode;
  }

  handleSubmit(values, actions) {
    const {
      setEditView, setNotificationMessage, updateProfileStore, updateProfile,
    } = this.props;
    const payload = this.createPayload(values);

    updateProfile(payload)
      .then(() => {
        setEditView(DEFAULT_VIEW);
        setNotificationMessage(USER_DETAILS_VIEW);
        actions.setSubmitting(false);
        updateProfileStore(payload);
      })
      .catch(() => {
        setNotificationMessage(USER_DETAILS_ERROR);
      });
  }

  createPayload(values) {
    const {
      title,
      firstName,
      lastName,
      dialingCountryCode,
      line1,
      line2,
      line3,
      line4,
      type,
      postCode,
      countryCode,
      companyName,
    } = values;

    const {
      profile,
      countries,
    } = this.props;

    const { dialingCode } = countries.find(country => (
      dialingCountryCode === country.countryCode));

    let mobile = '';
    if (values.mobile) {
      mobile = `${dialingCode}${values.mobile}`;
    }

    let telephone = '';
    if (values.telephone) {
      telephone = `${dialingCode}${values.telephone}`;
    }

    return {
      ...profile,
      contactDetail: {
        ...profile.contactDetail,
        title,
        firstName,
        lastName,
        mobile,
        telephone,
        address: {
          ...profile.contactDetail.address,
          line1,
          line2,
          line3,
          line4,
          postCode,
          countryCode,
          type,
          companyName: type === BUSINESS ? companyName : null,
        },
      },
    };
  }

  render() {
    const {
      dictionary,
      profile,
      setEditView,
      countries,
      app: { notificationMessage },
    } = this.props;

    const {
      title,
      firstName,
      lastName,
      email,
      address: {
        line1,
        line2,
        line3,
        line4,
        postCode,
        countryCode,
        type,
        companyName,
      },
    } = profile.contactDetail;


    const { mobile, telephone, dialingCountryCode } = this;

    return (
      <BaseStyle component="section" className="component-container">
        {notificationMessage === USER_DETAILS_ERROR && (
          <Notification data-test="error-notification" priority={2} variant="error" ariaLive="polite" show>
            {dictionary['generic.error']}
          </Notification>
        )}
        <Typography component="h4" className="pi-color-greyscale-darkest">
          {dictionary['profile.edit.title']}
        </Typography>
        <Typography component="h5" className="regular">
          {email}
        </Typography>

        <Formik
          data-test="user-profile-form-wrapper"
          onSubmit={this.handleSubmit}
          initialValues={{
            title,
            firstName,
            lastName,
            email,
            dialingCountryCode,
            mobile,
            telephone,
            countryCode,
            line1,
            line2,
            line3,
            line4,
            postCode,
            type,
            companyName,
            businessAddress: `${(type === BUSINESS)}`,
          }}
          validate={validateUserProfileForm}
          render={formikProps => (
            <UserProfileForm
              {...formikProps}
              dictionary={dictionary}
              setEditView={setEditView}
              profile={profile}
              countries={countries}
            />
          )}
        />
      </BaseStyle>
    );
  }
}

UserProfileFormContainer.propTypes = {
  setNotificationMessage: func.isRequired,
  app: object.isRequired,
  updateProfileStore: func.isRequired,
  dictionary: object.isRequired,
  profile: object.isRequired,
  countries: array.isRequired,
  setEditView: func.isRequired,
  updateProfile: func.isRequired,
};
