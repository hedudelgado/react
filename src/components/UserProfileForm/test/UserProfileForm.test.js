import React from 'react';
import { shallow } from 'enzyme';
import UserProfileForm from '../UserProfileForm';
import { DEFAULT_VIEW } from '../../App/appActions';

describe('UserProfileForm Component', () => {
  let wrapper;

  const props = {
    dictionary: {
      'profile.form.guesttitle.list': 'Mr,Mrs,Ms,Miss,Master,Dr,Lord,Lady,Sir,Col,Prof,Rev',
    },
    profile: {
      contactDetail: {
        title: 'Miss',
        firstName: 'Test',
        lastName: 'testing',
        email: 'test@whitbread.com',
        telephone: '11111111111',
        mobile: '+4900000000000',
        nationality: 'GB',
        passport: {
          number: '',
          countryOfIssue: '',
        },
        carRegistration: '',
        address: {
          line1: 'Random house',
          line2: 'Random street',
          line3: '',
          line4: 'LONDON',
          postCode: 'EC1N 3TD',
          countryCode: 'D',
          type: 'HOME',
          companyName: '',
        },
      },
    },
    application: {},
    touched: {},
    values: {
      dialingCountryCode: 'GB',
    },
    errors: {},
    countries: [
      {
        countryCode: 'GB',
        countryCodeISO: 'GB',
        countryLegend: 'United Kingdom (the)',
        passportRequired: false,
        dialingCode: '+44',
        flagImg: '/content/dam/global/flags/United-Kingdom.png',
      },
      {
        countryCode: 'D',
        countryCodeISO: 'DE',
        countryLegend: 'Germany',
        passportRequired: true,
        dialingCode: '+49',
        flagImg: '/content/dam/global/flags/Germany.png',
      },
      {
        countryCode: 'GR',
        countryCodeISO: 'GR',
        countryLegend: 'Greece',
        passportRequired: true,
        dialingCode: '+30',
        flagImg: '/content/dam/global/flags/Greece.png',
      },
      {
        countryCode: 'TEST',
        countryCodeISO: 'TEST',
        countryLegend: 'Test',
        passportRequired: true,
        dialingCode: '+499',
        flagImg: 'test.png',
      },
    ],
    setEditView: jest.fn(),
    handleBlur: jest.fn(),
    handleChange: jest.fn(),
    setFieldValue: jest.fn(),
    handleSubmit: jest.fn(),
  };

  beforeEach(() => {
    wrapper = shallow(<UserProfileForm {...props} />);
  });

  describe('Renders User Profile Form Component Elements', () => {
    it('should render the component', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should go back to default view on Cancel', () => {
      wrapper.find('[data-test="closeForm"]').simulate('click');
      expect(props.setEditView).toHaveBeenCalledWith(DEFAULT_VIEW);
    });
  });
});
