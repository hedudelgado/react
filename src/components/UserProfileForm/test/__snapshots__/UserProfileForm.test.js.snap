// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfileForm Component Renders User Profile Form Component Elements should render the component 1`] = `
<form
  data-test="user-profile-form"
  name="user-profile-form"
  onSubmit={[MockFunction]}
>
  <Select
    halfWidth={false}
    hideIcon={false}
    hideMessage={false}
    id="title"
    name="title"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    placeholder=""
  >
    <option
      key="Mr"
      value="Mr"
    >
      Mr
    </option>
    <option
      key="Mrs"
      value="Mrs"
    >
      Mrs
    </option>
    <option
      key="Ms"
      value="Ms"
    >
      Ms
    </option>
    <option
      key="Miss"
      value="Miss"
    >
      Miss
    </option>
    <option
      key="Master"
      value="Master"
    >
      Master
    </option>
    <option
      key="Dr"
      value="Dr"
    >
      Dr
    </option>
    <option
      key="Lord"
      value="Lord"
    >
      Lord
    </option>
    <option
      key="Lady"
      value="Lady"
    >
      Lady
    </option>
    <option
      key="Sir"
      value="Sir"
    >
      Sir
    </option>
    <option
      key="Col"
      value="Col"
    >
      Col
    </option>
    <option
      key="Prof"
      value="Prof"
    >
      Prof
    </option>
    <option
      key="Rev"
      value="Rev"
    >
      Rev
    </option>
  </Select>
  <Input
    id="firstName"
    mb="wb-push-middle--bottom"
    name="firstName"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    sessioncamhide={true}
    type="text"
  />
  <Input
    id="lastName"
    mb="wb-push-middle--bottom"
    name="lastName"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    sessioncamhide={true}
    type="text"
  />
  <Input
    disabled="disabled"
    id="email"
    mb="wb-push-middle--bottom"
    name="email"
    sessioncamhide={true}
    type="text"
  />
  <Select
    flagImg="/content/dam/global/flags/United-Kingdom.png"
    halfWidth={false}
    hideIcon={false}
    hideMessage={false}
    id="dialingCountryCode"
    name="dialingCountryCode"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    placeholder=""
    value="GB"
  >
    <option
      key="GB"
      label="+44"
      value="GB"
    >
      +44
    </option>
    <option
      key="D"
      label="Germany (+49)"
      value="D"
    >
      Germany (+49)
    </option>
    <option
      key="GR"
      label="Greece (+30)"
      value="GR"
    >
      Greece (+30)
    </option>
    <option
      key="TEST"
      label="Test (+499)"
      value="TEST"
    >
      Test (+499)
    </option>
  </Select>
  <Input
    id="mobile"
    mb="wb-push-middle--bottom"
    name="mobile"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    sessioncamhide={true}
    type="text"
  />
  <Input
    id="telephone"
    mb="wb-push-middle--bottom"
    name="telephone"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    sessioncamhide={true}
    type="text"
  />
  <AddressForm
    countries={
      Array [
        Object {
          "countryCode": "GB",
          "countryCodeISO": "GB",
          "countryLegend": "United Kingdom (the)",
          "dialingCode": "+44",
          "flagImg": "/content/dam/global/flags/United-Kingdom.png",
          "passportRequired": false,
        },
        Object {
          "countryCode": "D",
          "countryCodeISO": "DE",
          "countryLegend": "Germany",
          "dialingCode": "+49",
          "flagImg": "/content/dam/global/flags/Germany.png",
          "passportRequired": true,
        },
        Object {
          "countryCode": "GR",
          "countryCodeISO": "GR",
          "countryLegend": "Greece",
          "dialingCode": "+30",
          "flagImg": "/content/dam/global/flags/Greece.png",
          "passportRequired": true,
        },
        Object {
          "countryCode": "TEST",
          "countryCodeISO": "TEST",
          "countryLegend": "Test",
          "dialingCode": "+499",
          "flagImg": "test.png",
          "passportRequired": true,
        },
      ]
    }
    dictionary={
      Object {
        "profile.form.guesttitle.list": "Mr,Mrs,Ms,Miss,Master,Dr,Lord,Lady,Sir,Col,Prof,Rev",
      }
    }
    errors={Object {}}
    handleBlur={[MockFunction]}
    handleChange={[MockFunction]}
    profile={
      Object {
        "contactDetail": Object {
          "address": Object {
            "companyName": "",
            "countryCode": "D",
            "line1": "Random house",
            "line2": "Random street",
            "line3": "",
            "line4": "LONDON",
            "postCode": "EC1N 3TD",
            "type": "HOME",
          },
          "carRegistration": "",
          "email": "test@whitbread.com",
          "firstName": "Test",
          "lastName": "testing",
          "mobile": "+4900000000000",
          "nationality": "GB",
          "passport": Object {
            "countryOfIssue": "",
            "number": "",
          },
          "telephone": "11111111111",
          "title": "Miss",
        },
      }
    }
    setFieldValue={[MockFunction]}
    touched={Object {}}
    values={
      Object {
        "dialingCountryCode": "GB",
      }
    }
  />
  <Button
    className="wb-push--bottom"
    color="primary"
    data-test="save-changes"
    fullWidth={false}
    href={null}
    onClick={[Function]}
    submit={false}
    type="submit"
    variant="button"
  />
  <Button
    className="underline block"
    color="default"
    data-test="closeForm"
    fullWidth={true}
    href={null}
    onClick={[Function]}
    submit={false}
    variant="text"
  />
</form>
`;
